### ISSUE TITLE
Add support for {T-CONV:/Text/Type/} and {T-REPLACE-RX:/.../.../} as placeholder in fields

### ISSUE THREAD

**Issue by ArgusMagnus** [keepassxreboot/keepassxc#7293]
Following up on #1825, please consider adding support for {T-CONV:/Text/Type/} and {T-REPLACE-RX:/.../.../} as placeholder in fields. It's currently the feature I miss the most while switching over from KeePass.

My use-case:
Im trying to pass a base64-encoded value to a powershell script via the URL field:
`cmd://cmd /C "start powershell -ExecutionPolicy ByPass -File ^"{DB_DIR}\scripts\Test.ps1^" -Base64 {T-CONV:/{USERNAME}/Base64/}"`

The suggested workaround of manually converting the required value and storing it in a custom field does not work for us. We use field references (e.g. Username: `{REF:U@I:46C9B1FFBD4ABC4BBB260C6190BAD20C}`) and those cannot be used to retrieve custom fields.

**Comment by mark05e** [comment]
Upvoting this feature request. This would be useful.

**Comment by sh3bang** [comment]
Hello!

i have registered different mime types (using Linux) and defined different password entities using the URL field (e.g. http://,ssh://,sftp://,...) for each mime types.

When my password contains unescaped chars, the KeePassXC "URL Open" function don't work as expected.
Using KeePass2 this should work, using T-CONV function:
sftp://{USERNAME}:{T-CONV:/{PASSWORD}/Uri/}@{REF:A@I:3ee566cd69a84fb5a8c2624db97dcf6c}

**Comment by leo-liar** [comment]
Hello,

on work we recently switched from KeePass to KeePassXC and we've run into the same issue as described here and in #9504 – especially with regular expressions applied to the `{TITLE}` of some entries to specify (SAP) systems/clients: _(more or less exactly as the [inital comment in #9504](https://github.com/keepassxreboot/keepassxc/issues/9504#issue-1735059703))_

```batchfile
cmd://[..]sapshcut.exe -maxgui -system={T-REPLACE-RX:/{Title}/(?i)^(.* )?(\w+(?=(\s* \d+$)))\3/$2/} -client={T-REPLACE-RX:/{Title}/(?i)^.* (?=\d+$)//} -language=DE -user={USERNAME} -pw={PASSWORD}
```

(Which would resolve the URL for every entry with a title like `SAP [system] [client]` to `[..] -system=[system] -client=[client] [..]` so that different KeePass entries for different SAP systems would only need different titles but could all use the same URL command.)

I'm actually quite surprised that this is not supported yet (I was expecting feature parity to KeePass' functionalities) and according to #9504 also won't be supported in the future. Therefore I'd like to support this feature request and kindly ask to take it under consideration (again).

**Comment by droidmonkey** [comment]
I appreciate your problem but I don't know why this is the desired solution. Regexing a title to parse it into fields is not optimal, plus that regex is unreadable. You should just store the client and system information in additional attributes and use the `{S:attrname}` placeholder instead.

**Comment by leo-liar** [comment]
But then I would have to adjust about a dozen entries within the advanced tab and create (in our case) 2 special attributes for each instead of just adjusting the name.

We actually did share our SAP entry as template to other (not as much technically knowledgeable) users for its apparent ease – import the entry, adjust its name, username and password and done.

Readability of the regex aside, manually adding advanced information that's already in the title of an entry seems worse to me.

---

Edit: 
I actually tried to set the two needed attributes using the regexes but this didn't work either. (I was assuming that maybe `T-REPLACE-RX` just wasn't allowed within URL fields.)

**Comment by droidmonkey** [comment]
If you are using attributes, your url would look like this:

`cmd://[..]sapshcut.exe -maxgui -system={S:system} -client={S:client} -language=DE -user={USERNAME} -pw={PASSWORD}`

Case matters!

Adjusting a dozen entries would take you about 5 minutes max

**Comment by leo-liar** [comment]
Yeah, I set up the URL correctly (with correct case) but what I tried was to put the `T-REPLACE-RX` commands inside my manually configured attributes in hope they would work there but they get just input as is (without applying the regex substitution).

Adjusting all the entries for all users is what we will end up doing anyway, but it will most certainly take a lot longer than 5mins and is still a marked loss of functionality and usability in comparison to KeePass. Personally I'm using text conversions and regex replacements extensively in my database (not only for `sapgui`) and going on will lose a lot of automation (e.g changing only one entry which is referenced and regex-modified in place in other entries automatically) while using KeePassXC.

Again, even though it might be only a poweruser feature and not requested as much, I still urge you and the delevopers to reimplement said functionality. (I still don't quite understand the rationale of removing it / not even considering implementing it.)
